editor/src/components/inspector/common/longhand-shorthand-hooks.ts

```diff
@@ -245,14 +231,92 @@ export function useInspectorInfoLonghandShorthand<
       controlStyles: getControlStyles(controlStatus),
       onSubmitValue: onSubmitValue,
       onTransientSubmitValue: onTransientSubmitValue,
-      onUnsetValues: onUnsetValues,
+      onUnsetValues: NO_OP,
       orderedPropKeys: orderedPropKeys,
     }
   })
 
-  return mapArrayToDictionary(
+  const longhandResultsDictionary = mapArrayToDictionary(
     longhandResults,
     (_, index) => longhands[index],
     (result) => result,
   )
+
+  const longhandResultsWithUnset = objectMap((longhandResult, longhandToUnset) => {
+    const onUnsetValues = () => {
+      const allPropKeysEqual = allOrderedPropKeys.every((propKeys) => {
+        return arrayEquals(propKeys, allOrderedPropKeys[0])
+      })
+      if (!allPropKeysEqual) {
+        // we do nothing for now. we cannot ensure that we can make a sensible update and surface it on the UI as well
+        return
+      }
+      const propkeysToUse = allOrderedPropKeys[0]
+
+      const actionsToDispatch = (() => {
```

Comment:
Please can you pull out this inlined function and if possible break up the individual parts so it's easier to trace which actions will be fired under which conditions